### PR TITLE
Replace spelling wordlist sorting with Python

### PR DESCRIPTION
**Pull request by potiuk**
Part of #26020

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Not only replace the sorting with Python but also makes it case-insensitive (but stable) which makes it easier to find where to insert the word (I always struggled with casing there).

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -591,10 +591,11 @@ repos:
         require_serial: true
       - id: update-spelling-wordlist-to-be-sorted
         name: Sort alphabetically and uniquify spelling_wordlist.txt
-        entry: ./scripts/ci/pre_commit/pre_commit_sort_spelling_wordlist.sh
-        language: system
+        entry: ./scripts/ci/pre_commit/pre_commit_sort_spelling_wordlist.py
```
Comment (mik-laj) [inline .pre-commit-config.yaml:589]: Should we remove `./scripts/ci/pre_commit/pre_commit_sort_spelling_wordlist.sh` file?

**Comment by potiuk** [comment]
Removed :)

File: docs/spelling_wordlist.txt [hunk 2]
```diff
@@ -1594,24 +1577,41 @@ winrm
...
 yandexcloud
 yarnpkg
+Yieldr
 yml
 youtrack
 youtube
 zA
+Zego
+Zendesk
 zendesk
```
Comment (uranusjr) [inline docs/spelling_wordlist.txt:1610]: IIRC the spell checker is case insensitive so we should be able to get rid of these duplicates?

Comment (potiuk) [reply docs/spelling_wordlist.txt:1610]: Actually this wordlist is used for both Sphinx spellchecker and the codespell and I think the Sphinx one isn't case sensitive (and It's good as we can detect some casing that is wrong

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mik-laj) [reply .pre-commit-config.yaml:589]: @potiuk can you check if again?

Comment (potiuk) [reply .pre-commit-config.yaml:589]: Now removed for sure :)

**Review by mik-laj — APPROVED** [review]

**Pull request merged**